backend/FwLite/FwLiteProjectSync.Tests/Import/ResumableTests.cs

```diff
@@ -0,0 +1,211 @@
+                        ExampleSentences =
+                        [
+                            new ExampleSentence()
+                            {
+                                Id = Guid.NewGuid(), Sentence = { ["en"] = new($"example{i}", "en") }
+                            }
+                        ]
+                    }
+                ]
+            }).ToList();
+        var expectedPartsOfSpeech = Enumerable.Range(1, 10)
+            .Select(i => new PartOfSpeech { Id = Guid.NewGuid(), Name = { ["en"] = $"pos{i}" } }).ToList();
+
+        var mockFrom = new Mock<IMiniLcmApi>();
+        IMiniLcmApi mockTo = new UnreliableApi(
+            api: _fixture.Api,
+            random
+        );
+
+        mockFrom.Setup(f => f.GetWritingSystems()).ReturnsAsync(new WritingSystems()
+        {
+            Vernacular = [new()
+            {
+                Id = Guid.NewGuid(),
+                WsId = new() { Code = "en" },
+                Type = WritingSystemType.Vernacular,
+                Name = "English",
+                Abbreviation = "EN",
+                Font = "Arial",
+            }],
+            Analysis = [new()
+            {
+                Id = Guid.NewGuid(),
+                WsId = new() { Code = "en" },
+                Type = WritingSystemType.Analysis,
+                Name = "English",
+                Abbreviation = "EN",
+                Font = "Arial",
+            }]
+        });
+        mockFrom.Setup(f => f.GetPartsOfSpeech())
+            .Returns(MockAsyncEnumerable(expectedPartsOfSpeech));
+        mockFrom.Setup(f => f.GetEntries(It.IsAny<QueryOptions>()))
+            .Returns(MockAsyncEnumerable(expectedEntries));
+        mockFrom.Setup(f => f.GetPublications()).Returns(MockAsyncEnumerable([new Publication(){
+                Id = Guid.NewGuid(),
+                Name = { ["en"] = "Test Publication" },
+            }]));
+        mockFrom.Setup(f => f.GetComplexFormTypes())
+            .Returns(MockAsyncEnumerable([new ComplexFormType()
+            {
+                Id = Guid.NewGuid(),
+                Name = new(){ ["en"] = "Test Complex Form Type" }
+            }]));
+        mockFrom.Setup(f => f.GetSemanticDomains())
+            .Returns(MockAsyncEnumerable([new SemanticDomain()
+            {
+                Id = Guid.NewGuid(),
+                Name = new() { ["en"] = "Test Semantic Domain" },
+                Code = "TSD"
+            }]));
+
+        var import = new MiniLcmImport(
+            logger: NullLogger<MiniLcmImport>.Instance,
+            fwDataFactory: null!,
+            crdtProjectsService: null!
+        );
+
+        // Act: retry until all are imported
+        var maxTries = 20;
+        for (var attempt = 0; attempt < maxTries; attempt++)
+        {
+            try
+            {
+                await import.ImportProject(mockTo, mockFrom.Object, expectedEntries.Count);
+            }
+            catch (SimulatedException)
```

Comment:
Perhaps the test should assert that some simulated exceptions were thrown.

Reply:
except that it won't always throw. Eventually the import should finish and when it does it shouldn't throw.